packages/jest-haste-map/src/index.js

```diff
@@ -305,13 +305,28 @@ class HasteMap extends EventEmitter {
   /**
    * 2. crawl the file system.
    */
-  _buildFileMap(): Promise<InternalHasteMap> {
+  _buildFileMap(): Promise<{
+    deprecatedFiles: Array<{moduleName: string, path: string}>,
+    hasteMap: InternalHasteMap,
+  }> {
     const read = this._options.resetCache ? this._createEmptyMap : this.read;
 
     return Promise.resolve()
       .then(() => read.call(this))
       .catch(() => this._createEmptyMap())
-      .then(hasteMap => this._crawl(hasteMap));
+      .then(cachedHasteMap => {
+        const cachedFiles = Object.keys(cachedHasteMap.files).map(filePath => {
+          const moduleName = cachedHasteMap.files[filePath][H.ID];
+          return {moduleName, path: filePath};
+        });
+        return this._crawl(cachedHasteMap).then(hasteMap => {
+          const deprecatedFiles = cachedFiles.filter(file => {
+            const fileData = hasteMap.files[file.path];
+            return fileData == null || file.moduleName != fileData[H.ID];
```

Comment:
Just a reminder about strict equalities (sorry for being picky, but I just like it being more explicit and probably producing slightly lighter bytecode)

Reply:
Ugh, I added it and then refactored a bit and forgot again. We really should add a lint rule for this kind of things.

Reply:
Thanks! :)

Reply:
I'm keeping a simple `==` for the `null` though, as the rest of the file, because we don't want to make a distinction between `null` and `undefined`.

Reply:
I'm cool with that, let's be consistent.